### DISCUSSION TITLE
math.derivative returns a strange form of a polynomial involving negative powers.

### DISCUSSION THREAD

**Discussion by deSelbyPsyche** [josdejong/mathjs#3076] [Ideas]
**Describe the bug**
The bug is that when using math.derivative to evaluate `1x^2 -4x^-2`, mathjs returns: `(2 * x ^ 4 + 8) / x ^ 3`. I appreciate that this is one valid way of notating the answer, however, it is quite strange. An analogy would be that if an arithmetic function evaluated `2+2` and returned `8/2` we would likely consider that a bug. It is true that `8/2` is one way of notating the answer but it's not how the vast majority of users would expect to receive the result, which would be `4`.

It should be: 2 * x + 8 * x ^ -3, or `2 * x + 8 / x ^ 3`. The documentation includes some examples of derivatives, but not expressions like this, so maybe it was overlooked.

As a comparison the Python library SymPy evaluates `1*x**2 -4*x**-2`to: `2*x + 8/x**3`.

**To Reproduce**
Call math.derivative with the above polynomial and see that it evaluates to the above result.

**Comment by josdejong** [comment]
Makes sense. I'll label this as a possible improvement since it is not strictly speaking a bug.

Anyone interested in looking into this?

**Comment by deSelbyPsyche** [comment]
Hey thanks for getting back to my comment. If you think it's a discussion and not a bug since it is a correct answer to the question that's cool.

If I get more time I may come back to this and try to implement a change, but for now I have moved to another approach and don't have time.

**Comment by gwhitney** [comment]
As a sort of workaround at the moment,
`simplifyConstant(derivative("1x^2 -4x^-2", "x", {simplify: false}))` returns `2 * x ^ 1 - -8 * x ^ (-3)`, which is almost, but not quite, what you want. The issue is that `simplifyConstant` literally only simplifies the constant subexpressions; it can't change subtraction of a negative into adding a positive. The `simplify` function can and does, but it assumes that you would prefer a rational function of polynomials with positive exponents: note `simplify("2 * x ^ 1 - -8 * x ^ (-3)")` produces `(2 * x ^ 4 + 8) / x ^ 3`. So there are subtle questions here, resulting from the fact that there's no absolute definition of when one expression is more "simple" than another. Is `simplify` "wrong" to change `2x + 8x⁻³` to `(2x⁴+8)/x³`? Should only `rationalize` do that, and `simplify` leave it alone? Or, if we like polynomial/monomial for the result of `simplify`, then is there a different set of simplification rules that we should apply in the case of simplifying a derivative in particular? (In the end, I agree that the output of `derivative` in this case looks "odd" and that `2x + 8x⁻³` would be more "normal".)
